### PR TITLE
Infection state

### PR DESCRIPTION
**Pull request by alishbaimran**

### CODE REVIEW

File: applications/contrastive_phenotyping/graphs_ConvNeXt_ResNet.py [hunk 1]
```diff
@@ -1,4 +1,78 @@
 # %% Imports and paths.
+<<<<<<< HEAD
```
Comment (ziw-liu) [inline applications/contrastive_phenotyping/graphs_ConvNeXt_ResNet.py:2]: Merge error

**Review by ziw-liu — CHANGES_REQUESTED** [review]

File: applications/contrastive_phenotyping/graphs_ConvNeXt_ResNet.py [hunk 2]
```diff
@@ -1,4 +1,78 @@
...
+print(
+    f"shape of projections:{projections.shape}, shape of embedding: {embedding.shape}"
+)
+
+# %%
+plot_model = contrastive_resnet
+model_graph = torchview.draw_graph(
+    plot_model,
+    input_size=(20, in_channels, in_stack_depth, 224, 224),
+=======
```
Comment (ziw-liu) [inline applications/contrastive_phenotyping/graphs_ConvNeXt_ResNet.py:75]: merge error

File: viscy/data/hcs.py [hunk 3]
```diff
@@ -25,6 +25,8 @@
 from torch.utils.data import DataLoader, Dataset
 
 from viscy.data.typing import ChannelMap, DictTransform, HCSStackIndex, NormMeta, Sample
+import warnings
+warnings.filterwarnings("ignore", message="To copy construct from a tensor")
```
Comment (ziw-liu) [inline viscy/data/hcs.py:29]:
```suggestion
```
Don't filter this warning.

File: applications/contrastive_phenotyping/training_script.py [hunk 4]
```diff
@@ -150,64 +234,39 @@ def main(hparams):
...
+parser.add_argument("--embedding_len", type=int, default=256)
+parser.add_argument("--max_epochs", type=int, default=100)
+parser.add_argument("--accelerator", type=str, default="gpu")
+parser.add_argument("--devices", type=int, default=1)  # 4 GPUs
+parser.add_argument("--num_nodes", type=int, default=1)
+parser.add_argument("--log_every_n_steps", type=int, default=1)
+parser.add_argument("--num_workers", type=int, default=15)
+args = parser.parse_args()
+
+main(args)
```
Comment (ziw-liu) [inline applications/contrastive_phenotyping/training_script.py:271]: Will DDP work without a guard?

File: viscy/data/triplet.py [hunk 5]
```diff
@@ -165,22 +166,22 @@ def __getitem__(self, index: int) -> TripletSample:
...
+        initial_yx_patch_size: tuple[int, int] = (512, 512), # depends on affine transformations (roate and scaling), do full roation (sqrt(2)) and scaling (largest bounded box without padding).
+        final_yx_patch_size: tuple[int, int] = (224, 224), # 224, 224
+        split_ratio: float = 0.8, #training and prediction only 
         batch_size: int = 16,
         num_workers: int = 8,
-        normalizations: list[MapTransform] = [],
+        normalizations: list[MapTransform] = [], # same as VS except for target_channel
         augmentations: list[MapTransform] = [],
         caching: bool = False,
+        num_fovs: int = 4, # for quick testing
```
Comment (ziw-liu) [inline viscy/data/triplet.py:184]: Remove these comments

File: applications/contrastive_phenotyping/training_script.py (hunk 4 above)
Comment (alishbaimran) [reply applications/contrastive_phenotyping/training_script.py:270]: causes some issues

File: viscy/data/triplet.py [hunk 6]
```diff
@@ -266,7 +295,7 @@ def _setup_predict(self, dataset_settings: dict):
             tracks_tables=tracks_tables,
             initial_yx_patch_size=self.initial_yx_patch_size,
             anchor_transform=Compose(self.normalizations),
-            fit=False,
+            fit=False, #prediction dataset where fit is False
```
Comment (ziw-liu) [inline viscy/data/triplet.py:298]:
```suggestion
            fit=False,
```

File: viscy/light/engine.py [hunk 7]
```diff
@@ -712,13 +710,16 @@ def training_step(
...
         anchor = batch["anchor"]
         pos_img = batch["positive"]
         neg_img = batch["negative"]
-        emb_anchor = self.encoder(anchor)
-        emb_pos = self.encoder(pos_img)
-        emb_neg = self.encoder(neg_img)
-        loss = self.loss_function(emb_anchor, emb_pos, emb_neg)
+        _, anchorProjection = self.model(anchor)
+        _, negativeProjection = self.model(neg_img)
+        _, positiveProjection = self.model(pos_img)
```
Comment (ziw-liu) [inline viscy/light/engine.py:719]: Variables should be snake_case.

**Comment by ziw-liu** [comment]
Please refer to [the contributing guide](https://github.com/mehta-lab/VisCy/blob/main/CONTRIBUTING.md#ci-requirements) for CI requirements.

File: viscy/data/triplet.py [hunk 8]
```diff
@@ -173,14 +174,15 @@ def __init__(
...
-        final_yx_patch_size: tuple[int, int] = (256, 256),
+        initial_yx_patch_size: tuple[int, int] = (512, 512),
+        final_yx_patch_size: tuple[int, int] = (224, 224),
         split_ratio: float = 0.8,
         batch_size: int = 16,
         num_workers: int = 8,
         normalizations: list[MapTransform] = [],
         augmentations: list[MapTransform] = [],
         caching: bool = False,
+        num_fovs: int = 4,
```
Comment (ziw-liu) [inline viscy/data/triplet.py:185]: Is this needed in prod or just for debugging?

**Review by ziw-liu — APPROVED** [review]

**Pull request merged**